chanzuckerberg/single-cell-data-portal
# feat(wmg): performance improvements

## Reason for Change

- #4204 

## Changes

- add a new expression summary cube (`expression_summary_default`) that has been greatly simplified. the primary dimensions are gene, tissue, organism and the only secondary dimension is cell type. This cube is used for queries that do not specify any secondary filters.
- use the cell counts cube instead of the expression summary cube to determine the valid secondary filters given a set of primary filters.
- add timing logs to the response header of the query endpoint
- to help out functional performance tests, load the ontologies right off the bat at the global level instead of on the first function call
- set tile ordering to hilbert and capacity to 100,000 after empirical benchmarking

## Testing steps

- Tested and benchmarked in rdev.
- added a unit test to check that the default expression summary cube returns correct query results.
- added a unit test to ensure that both expression summary and expression summary default cubes have the same data in them (the latter is an aggregated version of the former).
